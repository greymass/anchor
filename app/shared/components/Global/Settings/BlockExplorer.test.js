import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';

import GlobalSettingsBlockExplorer from './BlockExplorer';

const mockProps = {
  actions: {
    setSetting: sinon.stub(),
  },
  blockExplorers: { 'Bloks IO': { url: 'bloks.io' } },
  name: 'Block Explorer',
};

describe('GlobalSettingsBlockExplorer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalSettingsBlockExplorer {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls setSetting is called when onchange is called', () => {
    const wrapper = mount(<GlobalSettingsBlockExplorer {...mockProps} />);

    wrapper.find('Dropdown').first().props().onChange(null, { value: 'Bloks IO' });
    expect(wrapper.props().actions.setSetting.calledWith('blockExplorer', 'Bloks IO')).toBeTruthy();
  });
});
