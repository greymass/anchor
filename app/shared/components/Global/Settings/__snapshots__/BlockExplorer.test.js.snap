// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalSettingsBlockExplorer renders correctly 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  deburr={false}
  defaultValue="Bloks IO"
  icon="dropdown"
  minCharacters={1}
  name="Block Explorer"
  noResultsMessage="No results found."
  onChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "text": "Bloks IO",
        "value": "Bloks IO",
      },
    ]
  }
  renderLabel={[Function]}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  wrapSelection={true}
/>
`;
